flask_rest_api/spec/__init__.py

```diff
@@ -132,13 +140,18 @@ def _register_redoc_rule(self, blueprint):
                     redoc_url = (
                         'https://cdn.jsdelivr.net/npm/redoc@'
                         '{}/bundles/redoc.standalone.js'.format(redoc_version))
-            self._redoc_url = redoc_url
+
```

Comment:
I'm not sure why you moved this function here. I think you could use `current_app.name` as in `_openapi_json`. Same thing goes for `openapi_swagger_ui` below.

Reply:
Thanks for reviewing.

I also need to pass `redoc_url`, `swagger_ui_url` and `swagger_ui_supported_submit_methods`.

I find it easier to use a closure than to pass them via the app object.

Or maybe I misunderstood your suggestion.

Are you suggesting I leave them as private attributes to the spec object (which always looked patchy to me, BTW) and I pass the spec object via the app object as `app.extensions['flask-rest-api']['spec']` ?